pkg/kcp/provider/gcp/vpcpeering/client/cloudComputeClient.go

```diff
@@ -105,9 +127,33 @@ func (c *networkClient) CreateVpcPeeringConnection(ctx context.Context, name *st
 	if err != nil {
 		return networkPeering, err
 	}
+	defer c.cnc.Close()
```

Comment:
defer will be run as soon as block finishes which means in the next line. So I am not sure what is the point of it. Could you please elaborate?